### PR TITLE
libertem UDF SSB implementation

### PR DESCRIPTION
**Pull request by lesan20**
## Contributor Checklist:

* [ ] I have added myself to [the creators.json file](https://github.com/Ptychography-4-0/ptychography/blob/master/packaging/creators.json)
* [ ] I have added [a changelog entry](https://github.com/Ptychography-4-0/ptychography/tree/master/docs/source/changelog) for my contribution
* [ ] I have added/updated documentation for all user-facing changes
* [ ] I have added/updated test cases

## Please remove this section

Starting by submitting an incomplete pull request (PR) or draft PR is OK. You
can work on the checklist step by step by pushing additional commits into the
PR. Please indicate if you think some items may not be applicable.

You can have a look at [our contributing
docs TODO update URL](https://libertem.github.io/LiberTEM/contributing.html) for more
information on contributing to LiberTEM. Please feel free to ask for
clarification and help, for example in your PR description.

Thank you for your contribution!

### CODE REVIEW

File: prototypes/ssb_mplementation/ssb.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+from libertem.udf import UDF  # NOQA: E402
+from libertem import api as lt  # NOQA: E402
+from libertem.executor.inline import InlineJobExecutor  # NOQA: E402
+
+from scipy.sparse import csr_matrix  # NOQA: E402
+
+
+def main():
+    path = r'C:\Users\lesnic\Nextcloud\Dieter\cGaN_sim_300kV\DPs\CBED_MSAP.raw'
```
Comment (SimeonEhrig) [inline prototypes/ssb_mplementation/ssb.py]: I believe, this line of code could cause a problem on other systems ;-)

Comment (lesan20) [reply prototypes/ssb_mplementation/ssb.py]: Yes, this is the first version for tests

**Comment by uellue** [comment]
From playing with the prototype a few thoughts:

* The bottleneck is the memory consumption of the mask stack when going to higher detector and image resolutions.
* Calculating individual mask layers is quite fast, as expected
* We could move from circles to arbitrary binary illumination masks by multiplying the mask with a shift of itself instead of comparing radii.

Instead of averaging already in the mask stack, we could use ones and zeros for the main reduction and deal with the weight of the individual channels in pre- or postprocessing. That would allow us to do tiled processing with masks that are calculated on the fly without having to know how many non-zero elements are in the mask. For that, we should collect the negative and positive contributions separately so that we can apply the average separately.

**Comment by uellue** [comment]
Merging as WIP to integrate CuPy support in a separate branch. @lesan20 you can just continue using your branch and open a new pull request for more changes!

**Pull request merged**